src/software/simulation/physics/physics_ball.h

```diff
@@ -84,15 +84,21 @@ class PhysicsBall
      * from its current location.
      *
      * @param in_flight_distance The distance for which the ball will be in flight
+     * @param angle_of_departure The angle of departure for the ball as it enters flight
      */
-    void setInFlightForDistance(double in_flight_distance);
+    void setInFlightForDistance(double in_flight_distance, Angle angle_of_departure);
 
     /**
      * Returns true if the ball is currently in flight, and false otherwise
      *
      * @return true if the ball is currently in flight, and false otherwise
      */
-    bool isInFlight();
+    bool isInFlight() const;
+
+    /**
+     * Updates whether or not the ball is "in flight" based on its current state.
+     */
+    void updateIsInFlight();
```

Comment:
Hmm, could we merge this back in to `isInFlight`? I wouldn't really expect to have to call this before `isInFlight`, and really wouldn't expect this to have any real side effects?

Reply:
Ah so I investigated this and found out why having it all in the same function doesn't/didn't work.

My assumption was that having `isInFlight()` also update its internal state tracking if the ball is in flight was fine, because it only needs to be updated to be "correct" when consumers need to know about it.

However part of the condition for being "no longer in flight" is that the ball is **not** touching another object (which helps detect if we "land on" another object). Currently the only consumer of `isInFlight()` is the `ContactListener`. So what happens is that
* ball gets chipped
* isInFlight() is true
* isInFlight() technically remains true even after the ball has passed the location it should have landed, because the ball landed in open space and so no contact callback occured, meaning `isInFlight()` was not called
* The ball later runs into another object
* The contact listener callback is now triggered to call `isInFlight()`, **but the ball is now touching another object again**, so `isInFlight()` still reports `true` which is incorrect

So I don't think we can recombine these functions while keeping proper behavior. What would make it better / clearer to help avoid having a function to update something, and a different function to get the result?

Reply:
Ah, makes sense. Could we call `updateIsInFlight` at the start of `isInFlight`? Then it should work and we don't need to explain any preconditions on calling `isInFlight`?

Reply:
Yes I can do that, but to be clear will still have to call `update` in the physics world update function